CONTRIBUTING.md

```diff
@@ -0,0 +1,236 @@
+# Contributing Guide
+
+This document is for anyone wanting to contribute to the implementation of the security tracker.
+
+## Overview
+
+This file contains general contribution information, but the other directories in this repository have additional `README.md` files with more specific information relevant to their sibling files:
+
+- [`src/website/`](./src/website/README.md): Sources and documentation for the web tracker.
+- [`nix`](./nix/README.md): Nix expressions for the service, tests, and the pkgs overlay.
+- [`pkgs`](./pkgs/README.md): Nix derivations for vendored Python dependencies.
+- [`contrib`](./contrib/README.md): Useful scripts for developing the project.
+- [`npins`](./npins/README.md): Pinned sources.
+- [`staging`](./staging/README.md): System configurations and service definitions for deployment.
```

Comment:
Tending to duplicate information is error-prone and increases friction when moving things around. Ideally we'd generate the table of contents automatically on each commit, but it's not wirth doing it now. I suggest that instead just say that each directory has a README explaining what it's about, and both reduce the number of directories and improve their legibility by organizing the code in a more obvious fashion.

For instance, IMO all the Nix-related support stuff, including `npins`, should just go into `nix`; `staging` should be called `deployment` -- if we need it at all; why would you store any machine-specifics in this repo? all you need are good defaults and module documentation -- and we probably don't need 3 Django applications either.